app/src/main/java/com/nutomic/syncthingandroid/service/EventProcessor.java

```diff
@@ -117,9 +121,16 @@ public void onEvent(Event event) {
 
                 notify(title, pi);
                 break;
+            case "FolderCompletion":
+                deviceId = (String) event.data.get("device");
+                folderId = (String) event.data.get("folder");
+                CompletionInfo completionInfo = new CompletionInfo();
+                completionInfo.completion = (Double) event.data.get("completion");
```

Comment:
Why can't we decode json straight into the struct, and then just updateCompletion(completion) and internally access folder and device id fields to do whatever is needed.

Reply:
Thats because of the structure difference describen in https://docs.syncthing.net/rest/db-completion-get.html and in https://docs.syncthing.net/events/foldercompletion.html . I created the CompletionInfo class according to the first source, so we can later improve the code and initially poll for it. To save resources, I decided to leave this part out as syncthing automatically emits this information when a device connects for the folder associated with it. If there are problems in the future that the event does not suffice, we have the full structure including globalBytes, needBytes, needDeletes available.
Btw, the event has just three fields in "event.data". An alternative would be to define an update structure separately (do we need this complexity at first integration?).

Reply:
I don't even see us calling /completion, so I don't see why should we follow the completion endpoint if it's never called.

Reply:
We'll need it in the future to display more info than just percentage.

Reply:
GUI calls it only after config save (for god knows what reason, probably legacy and can be removed), so I am not sure you will need it.

Reply:
For now, I added a comment in place and commented the unused values out.